[jovkusuma] iP

jovkusuma opened the pull request:
##  No Name Yet ChatBot

> "The only way to do great work is to love what you do." - Steve Jobs [source](https://www.goodreads.com/quotes/772887-the-only-way-to-do-great-work-is-to-love)

ChatBot frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. Download it from [here](https://github.com/jovkusuma/ip/releases/tag/A-Jar).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

**Features:**

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jovkusuma committed:
Level 0. Rename, Greet, Exit

jovkusuma committed:
Level 1. Echo

jovkusuma committed:
Level 2. Add, List

jovkusuma committed:
Level 3. Mark as Done

jovkusuma committed:
Level 4. ToDos, Events, Deadlines

jovkusuma committed:
A-TextUiTesting: Automated Text UI Testing

jovkusuma committed:
Level 5. Handle Errors

jovkusuma committed:
Level 6. Delete

jovkusuma committed:
Fix code structure

jovkusuma committed:
Implement storage file

jovkusuma committed:
Merge branch 'branch-Level-7'

jovkusuma committed:
Use LocalDate for time

jovkusuma committed:
Fix code errors

jovkusuma committed:
Segment code to different classes

jovkusuma committed:
Fix count mistake

jovkusuma committed:
Organize into packages

jovkusuma committed:
Remove comments in Duke file

jovkusuma committed:
Merge branch 'add-gradle-support'

jovkusuma committed:
Add JUnit Tests

jovkusuma committed:
Create a jar file

jovkusuma committed:
Add JavaDocs for each class and method

jovkusuma committed:
Adjust code according to Coding Standard

jovkusuma committed:
Implement find command

jovkusuma committed:
Merge branch 'branch-A-CodingStandard'

jovkusuma committed:
Resolve conflicts

jovkusuma committed:
Resolve issues

jovkusuma committed:
Add CheckStyle

jovkusuma committed:
Merge pull request #2 from jovkusuma/branch-A-CheckStyle

Merge branch 'branch-A-CheckStyle'

jovkusuma committed:
Implement GUI

jovkusuma committed:
Fix exception error

jovkusuma committed:
Move initialization of Storage

zhyuhan commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,122 @@
+                String mark = task.substring(4, 5);
+                String description = task.substring(7);
+
+                if (type.equals("T")) {
```
Perhaps you can consider refactoring this to a `switch` statement.

zhyuhan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+    /**
+     * Runs the Duke chatbot.
+     */
+    public void run() {
```
Perhaps you can consider adding line breaks in the method body to separate logical blocks and improve readability.

zhyuhan commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+     * @throws DukeException If the command execution fails.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (command.equalsIgnoreCase("todo")) {
```
As a suggestion, may you could extract the lower-case command into a separate variable and use `switch` statements to handle the different cases instead.

zhyuhan commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (index >= 0 && index < tasks.getTaskCount()) {
```
Perhaps you can refactor this method to return early to reduce nesting.

zhyuhan commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,122 @@
+     * @param task the task to be marked as done
+     */
+    public void markDonePrint(Task task) {
+        System.out.println(LINE);
```
The `System.out.println(LINE)` before and after the output seems to be repeated in several places. Perhaps you could abstract it into a separate method to reduce code duplication.

zhyuhan reviewed (commented):
Overall really clean code, I had to really nitpick to find issues with coding quality. I also really like the structure of your program and the various levels of abstraction!

jovkusuma committed:
Implement bye and fix edge cases

jovkusuma committed:
Merge pull request #3 from jovkusuma/branch-Level-10

Merge branch 'branch-Level-10' to master

jovkusuma committed:
Fix code according to checkStyle

jovkusuma committed:
Fix code according to checkStyle

jovkusuma committed:
Change javafx version in build.gradle

jovkusuma committed:
Add Assertions

jovkusuma committed:
Refactor to improve code quality

jovkusuma committed:
Made slight changes to assertions

jovkusuma committed:
Turn task commands into individual methods

jovkusuma committed:
Merge pull request #4 from jovkusuma/branch-A-Assertions

Merge 'branch-A-Assertions' to master

jovkusuma committed:
Merge pull request #6 from jovkusuma/branch-A-CodeQuality

Merge 'branch-A-CodeQuality' to master

jovkusuma committed:
Fix wrong condition for assertions

jovkusuma committed:
Add assertions

The current program lacks assertion checks, which is used to define
assumptions about the program state so that the runtime can verify them.

There are several methods in TaskList that may not work as
intended when the input is invalid.

Using assertions can help users detect any input mistake.

Let's add some assertion checks to TaskList methods to check
the assumptions that should hold.

jovkusuma committed:
Merge pull request #7 from jovkusuma/branch-A-Assertions

Add assertions

jovkusuma committed:
Add shortcuts to different commands
The current program has fixed syntax which may not be suitable for
people who want to enter commands quickly.

Some users may not like typing in long commands to do a command

Made changes to the syntax of the command to be more flexible for
fast-typers

This change is made to be more flexible and enhance the overall
user experience

Additionally, there is also a new command "help" added to help users
view the list of commands and its syntax.

jovkusuma committed:
Merge pull request #9 from jovkusuma/branch-BCD-Extension

Merge 'branch-BCD-Extension' to master

jovkusuma committed:
Add User Guide

jovkusuma committed:
Edit user guide

jovkusuma committed:
Fix formatting of user guide

jovkusuma committed:
Fix help output
